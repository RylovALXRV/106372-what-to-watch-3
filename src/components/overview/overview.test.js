import React from "react";
import renderer from "react-test-renderer";
import Overview from "./overview.jsx";

const mock = {
  descriptions: [`In late 1823, Hugh Glass guides Captain Andrew Henry's trappers through territory of the present day Dakotas.`],
  rating: 8.0,
  director: `David Yates`,
  starring: [`Rami Malek`, `Jon Finch`, `Cate Blanchett`],
};

it(`Should Overview render correctly`, () => {
  const {descriptions, rating, director, starring} = mock;
  const tree = renderer.create(
      <Overview
        rating={rating}
        descriptions={descriptions}
        starring={starring}
        director={director}
      />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
