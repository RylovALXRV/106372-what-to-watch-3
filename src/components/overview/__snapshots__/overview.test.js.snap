// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Overview render correctly 1`] = `
Array [
  <div
    className="movie-rating"
  >
    <div
      className="movie-rating__score"
    >
      8
    </div>
    <p
      className="movie-rating__meta"
    >
      <span
        className="movie-rating__level"
      >
        Very good
      </span>
      <span
        className="movie-rating__count"
      >
        240 ratings
      </span>
    </p>
  </div>,
  <div
    className="movie-card__text"
  >
    <p>
      In late 1823, Hugh Glass guides Captain Andrew Henry's trappers through territory of the present day Dakotas.
    </p>
    <p
      className="movie-card__director"
    >
      <strong>
        Director: 
        David Yates
      </strong>
    </p>
    <p
      className="movie-card__starring"
    >
      <strong>
        Starring: 
        Rami Malek, Jon Finch, Cate Blanchett
         and other
      </strong>
    </p>
  </div>,
]
`;
